spec/rest/tls.test.js

```diff
@@ -0,0 +1,118 @@
+"use strict";
+
+define(['ably', 'shared_helper', 'async'], function(Ably, helper, async) {
+	var currentTime, exports = {};
+
+	/*
+	 * Set up and time-sync.
+	 */
+	exports.setupauth = function(test) {
+		test.expect(1);
+		helper.setupApp(function() {
+			var rest = helper.AblyRest();
+			(function(callback) {
+				rest.time(function(err, time) {
+					if(err) { callback(err); }
+					callback(null, time);
+				});
+			})(function(err, time) {
+				if(err) {
+					test.ok(false, helper.displayError(err));
+					test.done();
+					return;
+				}
+				currentTime = time;
+				test.ok(true, 'Obtained time');
+				test.done();
+			});
+		});
+	};
+
+	/*
+	 * tls:true initiates https connection
+	 */
+	exports.tls_connection = function(test) {
+		test.expect(4);
+
+		var rest,
+		keyStr = helper.getTestApp().keys[1].keyStr;
+
+		rest = helper.AblyRest({ tls: true, key: keyStr });
+
+		async.parallel([
+			function(cb) {
+				rest.auth.requestToken(function(err, tokenDetails) {
+					test.ok( tokenDetails.token, 'Verify that token is returned');
+					cb();
+				});
+			},
+			function(cb) {
+				rest.stats(function(err, stats) {
+					test.ok(err, 'Verify that an error was returned')
```

Comment:
> What has this to do with tls tests? Why is this bundled into this test?

I am quoting RSC18:

> (RSC18) If ClientOptions#tls is true, then all communication is over HTTPS. **If false, all communication is over HTTP** however Basic Auth over HTTP will raise an exception as private keys cannot be submitted over an insecure connection. See Auth below

I was testing the part in bold. My reasoning is that the case of `tls = false` should be tested to make sure that tls is not used when explicitly disabled. I thought I'd bundle it with other tls-related tests. Would you like me to move it to some other file or remove it?

> I am just not clear why you are testing a failing stats request?

Before starting to write these tests I made several attempts to force the client to send a secret key over unencrypted http. I was successful to do this using a stats request. Then I modified the code to make sure that this condition returns an error rather than sending the key unencrypted. Here I'm testing for this error. Did you have something else in mind?